### DISCUSSION TITLE
How can I deploy my Git repo, which has both an express backend and a react frontend?

### DISCUSSION THREAD

**Discussion by Josrph** [vercel/community#1423] [Help]
### Summary

How can I deploy my Git repo, which has both an express backend and a react frontend?

I tried deploying it directly, but I got `404: NOT_FOUND.`

the folder structure of my repo is like this:

```
- simple-react-express (main folder)
  - react (sub folder)
  - express (sub folder)
```

my front end is using a proxy to connect to my backend so I have in package.json: `"proxy": "http://localhost:5000"`, and my backend is running on the same address.

This is my first time ever using vercel, So what are the steps I should follow to successfully deploy?

### My Git Repo
https://github.com/Josrph/simple-react-express

**Comment by amyegan** [comment]
Hi Joseph! Vercel is designed for hybrid frontends and lightweight serverless APIs. Because the environment is serverless, rather than a traditional server, you'll need to move your Express code into a Serverless Function. 

The [using Express.js with Vercel](https://vercel.com/guides/using-express-with-vercel) guide covers steps to implement serverless Express as well as some drawbacks to consider.

**Reply by Josrph** [reply]
After some trial and error, I was able to deploy both the frontend and the backend separately, and I didn't have to move my Express code into a Serverless Function like you said, but, although they both work fine on their own, for some reason any API calls to the backend using axios don't work.

So now the real question is, how can I debug my React frontend once deployed to vercel or is there any way to simulate vercel runtime environment locally, so I can find out what's not working?

**Reply by amyegan** [reply]
[Vercel CLI](https://vercel.com/docs/cli#) can help with that. You can use [`vercel dev`](https://vercel.com/docs/cli/dev#) to imitate the Vercel deployment environment locally

**Reply by Josrph** [reply]
Unfortunately, after trying the `vercel dev` command, I found that everything works fine locally, any API calls to vercel backend using axios from my local frontend works fine, the issue only occurs when actully deploying to vercel.

Somehow axios cannot connect to the backend and I get an axios exception `Network Error`, I only found out after replacing the `cnoslog.log` function call with `alert` in the try catch block.

So now am back to square one, do you have any ideas on what to do next?

**Reply by amyegan** [reply]
I think you're going to need to move your Express code into an `/api` directory to deploy it as a Serverless Function, like in the [Express.js with Vercel](https://vercel.com/guides/using-express-with-vercel) guide. You might also need to take a look at [adding CORS headers](https://vercel.com/guides/how-to-enable-cors) or [rewrites](https://vercel.com/docs/concepts/projects/project-configuration#rewrites) to allow communication across domains/subdomains

**Reply by srihernogo** [reply]
> After some trial and error, I was able to deploy both the frontend and the backend separately, and I didn't have to move my Express code into a Serverless Function like you said, but, although they both work fine on their own, for some reason any API calls to the backend using axios don't work.
> 
> So now the real question is, how can I debug my React frontend once deployed to vercel or is there any way to simulate vercel runtime environment locally, so I can find out what's not working?

Hi @Josrph, I have same problem with your, please tell how you can solve this issue...
Thank you
